Remove unused utility methods checking for Java 8

Chrimle opened the pull request:
Given the fact that JVM versions were prefixed with '1.' until Java 9, there are 2 helper methods regarding the Java version being used; 'Java 8 or older' and 'Java 9 or newer'.

Since the methods are mutually exclusive and are essentially an inverse of one another, the issue was that both methods get the runtime Java version and then check if the version begins with '1.'. But because of the inverse property, only one method needs to do this check, and the other call the first and invert the result.


<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8152"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

Chrimle committed:
Evaluate if JVM is Java 9+ by inverting 'Java 8 or older' check

Given the fact that JVM versions were prefixed with '1.' until Java 9,
there are 2 helper methods regarding the Java version being used; 'Java 8 or older' and 'Java 9 or newer'.

Since the methods are mutually exclusive and are essentially an inverse of one another, the issue was that both methods get the runtime Java version and then check if the version begins with '1.'. But because of the inverse property, only one method needs to do this check, and the other call the first and invert the result.

mawinter69 commented:
Considering that Jenkins is built with java11 you could simply return `true` for isRunningWithPostJava8 and `false` for isRunningWithJava8OrBelow
That said these methods seems to be obsolete somehow maybe they should be deprecated together with the JavaVersionRecommendationAdminMonitor for now. At a later point in time it might make sense to reactivate this when Jenkins wants to recommend using java 17 maybe.

NotMyFault commented:
> That said these methods seems to be obsolete somehow maybe they should be deprecated

That'd make sense, although that'd be something for a different PR, to cleanup the Java 8 warning stuff.

Hey @Chrimle,

do you think you can apply the simplification recommended?

Chrimle commented:
Hi @NotMyFault & @mawinter69 !

Sure thing, I could to the suggested changes. Should I at the same time also mark these as deprecated?

Chrimle committed:
Simplify methods & Mark as Deprecated

Suggestion was to simplify the methods, as the Java version is always 11 or newer.

For that reason, it was also suggested to mark these methods deprecated - and perhaps be replaced by similar util-methods for checking "if java >17"

Chrimle commented:
Is there a specific 'since' version that should be referenced instead of a date?

Description of Deprecation;

> @**deprecated** because the current version is at least Java 11, this method is redundant.

Chrimle committed:
Added Deprecated to JavaDocs

Added deprecated tag to JavaDoc due to checkStyle fail in CI, along with short explanation for deprecation.

NotMyFault commented:
> Is there a specific 'since' version that should be referenced instead of a date?

TODO; automatic tooling will fill in the tag version it has been released in.

Chrimle commented:
> `TODO`; automatic tooling will fill in the tag version it has been released in.

Oh, very cool! Will update this. Thanks!

Chrimle committed:
Corrected Deprecated 'since'

Chrimle committed:
Merge branch 'master' into patch-1

Chrimle committed:
Fix deprecated 'since' in JavaVersionRecommendationAdminMonitor

Chrimle committed:
Merge branch 'master' into patch-1

NotMyFault commented:
Just FYI, you need to mark this PR as "Ready for review" over draft, to get reviews 👀

timja commented on core/src/main/java/jenkins/monitor/JavaVersionRecommendationAdminMonitor.java:
```diff
@@ -44,6 +44,7 @@
 @Extension
 @Restricted(NoExternalUse.class)
 @Symbol("javaVersionRecommendation")
+@Deprecated(since = "TODO", forRemoval = true)
```
This is going to be re-introduced soon, I don't think there's a point in removing it right now.

MarkEWaite committed:
Merge branch 'master' into patch-1

MarkEWaite committed:
Removal deprecation of JavaVersionRecommendationAdminMonitor

MarkEWaite committed:
Simplify test and update Javadoc comment for current status

Java 25 support is coming and Java 11 is no longer supported.

Chrimle reviewed (commented):
The new additions looks great 👍

timja commented on core/src/main/java/jenkins/util/java/JavaUtils.java:
```diff
@@ -43,19 +43,23 @@ private JavaUtils() {
+     * this method is redundant.
      */
+    @Deprecated(since = "TODO", forRemoval = true)
     public static boolean isRunningWithPostJava8() {
```
Neither of these methods are used in jenkinsci 

https://github.com/search?q=org%3Ajenkinsci+isRunningWithPostJava8&type=code

https://github.com/search?q=org%3Ajenkinsci%20isRunningWithJava8OrBelow&type=code

Can we just remove them?

MarkEWaite replied:
Thanks for asking the question.  I confirmed with [usage in plugins](https://www.jenkins.io/doc/developer/plugin-development/usage-in-plugins/) that those two methods are unused in Jenkins plugins.  I also searched the cloudbees organization on GitHub and confirmed they are unused there as well.

Removed in ed9b418c499e37435a79845c5220dc81f7589c93

MarkEWaite committed:
Merge branch 'master' into patch-1

MarkEWaite committed:
Remove isRunningWithJava8OrBelow and isRunningWithPostJava8 from JavaUtils

Usage in plugins shows that those methods are unused.  No usages were
found in either the jenkinsci GitHub organization or the cloudbees
GitHub organization.
